pubtools/exodus/_tasks/push.py

```diff
@@ -0,0 +1,66 @@
+import logging
+import subprocess
+
+from pubtools.exodus.task import ExodusTask
+
+LOG = logging.getLogger("pubtools-exodus")
+LOG_FORMAT = "%(asctime)s [%(levelname)-8s] %(message)s"
+
+
+class ExodusPushTask(ExodusTask):
+    """Push a directory to the Exodus CDN"""
+
+    def add_args(self):
+        super(ExodusPushTask, self).add_args()
+
+        self.parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            help="Perform a trial run with no changes made",
+        )
+        self.parser.add_argument(
+            "--source",
+            help="Local path to a file or directory for sync to the exodus CDN",
+        )
+        self.parser.add_argument(
+            "--dest",
+            help="Remote destination for sync",
+        )
+
+    def run(self):
+        LOG.debug("Exodus push begins")
+
+        src = self.args.source
+        dest = (
+            self.args.dest
+            if ":" in self.args.dest
+            else "exodus:%s" % self.args.dest
+        )
+
+        cmd = [
+            "exodus-rsync",
+            src,
+            dest,
+        ]
+        if self.args.verbose:
+            cmd.append("-" + "v" * self.args.verbose)
+        if self.args.dry_run:
+            cmd.append("--dry-run")
+
+        LOG.info(" ".join(cmd))
+
+        proc = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT
+        )
+        for line in proc.stdout:
+            LOG.info(line.decode("utf-8").strip())
```

Comment:
Minor - if you include universal_newlines=True in the arguments to Popen, it should decode the bytes for you automatically.  (argument name is 'text' on modern python, but 'universal_newlines' has to be used for py2 compat)

Reply:
Thank you!